conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowServiceHandlerGenerator.java

```diff
@@ -559,7 +577,22 @@ private static CodeBlock generateParamMetadata(
                                 .withoutAnnotations()
                                 .toString()),
                         argument.getTags().stream())
+                // filter out both forms of 'safe' in favor of the declared
+                .filter(item -> !Safe.class.getName().equals(item) && !"safe".equals(item))
                 .collect(ImmutableSet.toImmutableSet());
+        // Body parameters can very noisy, we do not attach these by default.
+        if (!argument.getParamType().accept(ParameterTypeVisitor.IS_BODY)
+                && !mergedTags.contains("server-squelch")
+                && safetyEvaluator
+                        .evaluate(argument.getType(), ConjureTags.safety(argument))
+                        .filter(LogSafety.SAFE::equals)
+                        .isPresent()) {
+            mergedTags = ImmutableSet.<String>builder()
+                    .add("safe")
+                    .addAll(mergedTags)
+                    .build();
+        }
```

Comment:
Longer term it would make sense to define a new API backed by the same thing as `requestContext.requestArg`, but I opted for a minimal change.

Reply:
I bit the bullet and updated codegen to use the RequestContext, matching the annotation processor :-]
I think it makes the generated code a bit more natural.

Reply:
Agree, the requestContext format looks more readable to me!